core/helpers_test.go

```diff
@@ -1,11 +1,11 @@
 package core
 
 import (
-	"testing"
 	"reflect"
+	"testing"
 )
 
-func checkErrors(t *testing.T, name string, index int, err error, expected error) {
+func checkErrors(t *testing.T, name string, index int, err Errs, expected Errs) {
```

Comment:
Because it's used `reflect.DeepEqual` the parameters wouldn't need to change and the function will be more generic for any test.  Being a test we don't need to show any semantics in docs which types can provide